config/nginx/dice-staging.tripleawarclub.org

```diff
@@ -1,3 +1,7 @@
+# Session caching for improved performance
+
+ssl_session_cache shared:ssl_session_cache:10m;
```

Comment:
nginx basically includes all files in the sites-enabled directory, therefore this setting would be applied twice, but I don't think this causes a problem, can't guarantee it though.

Reply:
Actually, won't this setting be applied _three_ times since the _tripleawarclub.org_ configuration is also part of the same Nginx instance?

Since we know all of these virtual servers are running in the same instance, would it make sense to just say that _tripleawarclub.org_ will contain the global `http` configuration for all servers?  That is, remove `ssl_session_cache` from all configurations except _tripleawarclub.org_?

Reply:
Right.
It would actually be more consistent to just put it directly into the global nginx config at `/etc/nginx/nginx.conf`, just above the include statement for the individual configs, but I hesitated to edit the config directly, because this may make updating nginx annoying when the original configuration gets changed.

Reply:
Agreed.  Do you want me to add _nginx.conf_ to the `tripleawarclub.org` repo (for lack of a better location) and instructions for updating it to the README?

Reply:
Just when thinking about it:
In theory it should just work to create a new file (which then gets automatically included) which just contains this single instruction and nothing else.
This way we don't need to fiddle around with the config and still get the same result, without any potential weird side effects.

Reply:
If we do that, then it looks like you could just remove `TLSv1` from the global configuration instead of excluding it for each server.  Would that be equivalent to changing all three server configurations?

Reply:
Yup that would be equivalent, still not sure why this setting can be inside as well as outside of server blocks.
I'll not be able to come back at this for the next hour, so if you want to do this simple change, feel free to either close this PR (and the other one) and to create your own, or to push to the remote branches of either of them

Reply:
It looks like all the `ssl_*` directives we use can be applied either at `http` or `server` scopes.  May want to keep that in mind if we go with a global TLS configuration for all servers.

Reply:
To summarize this discussion...  It sounds like you're proposing to add a separate configuration that effectively represents the common configuration for all servers, correct?

If so...

1. Should that file be hosted in the _tripleawarclub.org_ repo?
1. Should it be deployed to _sites-enabled_ or would it be less confusing if it were deployed to _conf.d_?  It seems files from both folders are included by _nginx.conf_.
1. Should we move all `ssl_*` configuration to this file now or just what you're proposing in these PRs?

Reply:
Yes,
1. Not sure, this repo is only being kept because we store the config to do the redirect there. So creating a separate repo or using the infrastructure repo might be a better choice.
2. Didn't know conf.d is included as well. If it works it's probably better 👍
3. Not all of them, we probably want to keep settings that specify a path of some sort seperate, but in general yes.

Got a bit late for me, so I won't be available for at least the next 12 hours.
Going to close both PRs as we agreed that there are fundamentally better ways of managing this all.

Reply:
No problem.  I'll probably have more time tomorrow anyway if you still need me to deploy it. :smile:

Reply:
Sorry, no stable Internet today, so not able to create a PR :/

Reply:
Ok, back to this topic again.
Unfortunately the infrastructure repo doesn't has a "store only" folder for configuration files.
If you have a good suggestion on where to put them I'd be happy to hear it.
Unless you think it'd be worth it to directly incorporate the dice server into the infrastructure process (I don't think it'll be worth it)

Reply:
Sorry, I don't have a good suggestion either.  The only thing I'd recommend is, if you put it in the `infrastructure` repo, you put it in a top-level folder that makes it clear it's not really part of the current infrastructure process.  For example, _other_, _etc_, _todo_, etc. (none of those are really that good).  A README in that folder with one or two sentences stating that purpose would be helpful.

I agree that it's not worth it right now to spend time incorporating MARTI into infrastructure given how rarely it gets updated.